src/train_cmd.cpp

```diff
@@ -444,6 +444,9 @@ int Train::GetDisplayImageWidth(Point *offset) const
 
 	if (offset != NULL) {
 		offset->x = ScaleGUITrad(reference_width) / 2;
+		if (HasBit(this->flags, VRF_REVERSE_DIRECTION)) {
+			offset->x -= (VEHICLE_LENGTH - this->gcache.cached_veh_length) * reference_width / VEHICLE_LENGTH;
```

Comment:
This line only works when GUI zoom is set to 1x.
Make sure that you use
offset->x = ScaleGUITrad(...);
and do all stuff related to cached_veh_length within the "...", or first compute the unscaled position and then call ScaleGUITrad one.

Reply:
Can you also please test the behaviour if you switch to a language with right-to-left text, like arabic or hebrew.
IIRC the trains switch direction in depot for those languages, and this may affect the offsets here.